WebCryptoAPI/wrapKey_unwrapKey/wrapKey_unwrapKey.https.any.js

```diff
@@ -6,239 +6,249 @@
+    async function wrapAsNonExtractableJwk(key, wrapper) {
         var wrappingKey = wrapper.wrappingKey,
             encryptKey;
 
-        return subtle.exportKey("jwk",wrappingKey)
-        .then(function(jwkWrappingKey){
-            // Update the key generation parameters to work as key import parameters
-            var params = Object.create(wrapper.parameters.generateParameters);
-            if(params.name === "AES-KW") {
-                params.name = "AES-CBC";
-                jwkWrappingKey.alg = "A"+params.length+"CBC";
-            } else if (params.name === "RSA-OAEP") {
-                params.modulusLength = undefined;
-                params.publicExponent = undefined;
-            }
-            jwkWrappingKey.key_ops = ["encrypt"];
-            return subtle.importKey("jwk", jwkWrappingKey, params, true, ["encrypt"]);
-        }).then(function(importedWrappingKey){
-            encryptKey = importedWrappingKey;
-            return subtle.exportKey("jwk",key);
-        }).then(function(exportedKey){
-            exportedKey.ext = false;
-            var jwk = JSON.stringify(exportedKey)
-            if (wrappingKey.algorithm.name === "AES-KW") {
-                return aeskw(encryptKey, str2ab(jwk.slice(0,-1) + " ".repeat(jwk.length%8 ? 8-jwk.length%8 : 0) + "}"));
-            } else {
-                return subtle.encrypt(wrapper.parameters.wrapParameters,encryptKey,str2ab(jwk));
-            }
-        });
+        var jwkWrappingKey = await subtle.exportKey("jwk",wrappingKey);
+        // Update the key generation parameters to work as key import parameters
+        var params = Object.create(wrapper.parameters.generateParameters);
+        if(params.name === "AES-KW") {
+            params.name = "AES-CBC";
+            jwkWrappingKey.alg = "A"+params.length+"CBC";
+        } else if (params.name === "RSA-OAEP") {
+            params.modulusLength = undefined;
+            params.publicExponent = undefined;
+        }
+        jwkWrappingKey.key_ops = ["encrypt"];
+        var importedWrappingKey = await subtle.importKey("jwk", jwkWrappingKey, params, true, ["encrypt"]);
+        encryptKey = importedWrappingKey;
+        var exportedKey = await subtle.exportKey("jwk",key);
+        exportedKey.ext = false;
+        var jwk = JSON.stringify(exportedKey)
+        var result;
+        if (wrappingKey.algorithm.name === "AES-KW") {
+            result = await aeskw(encryptKey, str2ab(jwk.slice(0,-1) + " ".repeat(jwk.length%8 ? 8-jwk.length%8 : 0) + "}"));
+        } else {
+            result = await subtle.encrypt(wrapper.parameters.wrapParameters,encryptKey,str2ab(jwk));
+        }
+        return result;
     }
 
 
     // RSA-OAEP can only wrap relatively small payloads. AES-KW can only
     // wrap payloads a multiple of 8 bytes long.
     function wrappingIsPossible(exportedKey, algorithmName) {
+        if (exportedKey === undefined) {
+            return false
+        }
```

Comment:
Are there legitimate cases where the `exportedKey` is not available here, or is this perhaps hiding the issue? (I.e. perhaps there's a race condition where sometimes it's not available yet? Although I don't quite see how that could happen..)

Reply:
That case is for the map that contains the exportedKeys indexed by format; undefined there means that the format was not valid for the export operation (eg, "pkcs8" and public-key)